parity-crypto/src/publickey/ecdsa_signature.rs

```diff
@@ -0,0 +1,316 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! ECDSA signature
+
+use std::ops::{Deref, DerefMut};
+use std::cmp::PartialEq;
+use std::fmt;
+use std::str::FromStr;
+use std::hash::{Hash, Hasher};
+use secp256k1::{Message as SecpMessage, RecoverableSignature, RecoveryId, Error as SecpError};
+use secp256k1::key::{SecretKey, PublicKey};
+use rustc_hex::{ToHex, FromHex};
+use ethereum_types::{H520, H256};
+use super::{Secret, Public, SECP256K1, Message, public_to_address, Address, Error};
+
+/// Signature encoded as RSV components
+#[repr(C)]
+pub struct Signature([u8; 65]);
+
+impl Signature {
+	/// Get a slice into the 'r' portion of the data.
+	pub fn r(&self) -> &[u8] {
+		&self.0[0..32]
+	}
+
+	/// Get a slice into the 's' portion of the data.
+	pub fn s(&self) -> &[u8] {
+		&self.0[32..64]
+	}
+
+	/// Get the recovery byte.
+	pub fn v(&self) -> u8 {
+		self.0[64]
+	}
+
+	/// Encode the signature into RSV array (V altered to be in "Electrum" notation).
```

Comment:
"Electrum" as in the bitcoin wallet? Can you elaborate on what we use this for? And perhaps bake it into the comment here?

Reply:
This history is pretty vague already. Such format was mentioned in the yellow paper: https://ethereum.stackexchange.com/questions/55241/what-is-electrum-style-signature?rq=1 but why it was used and for which cases... it's already hidden in the mist (at least for me)

Reply:
:/

Seems like it's used in `rpc` and `private-tx` (and tests).